airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/generate_erd/pipeline.py

```diff
@@ -0,0 +1,266 @@
+#
+from __future__ import annotations
+
+import json
+import os
+from pathlib import Path
+from typing import TYPE_CHECKING, List, Optional
+
+import dpath
+import google.generativeai as genai
+
+from airbyte_protocol.models import AirbyteCatalog
+from dagger import Container
+from markdown_it import MarkdownIt
+from pipelines.airbyte_ci.connectors.build_image.steps.python_connectors import BuildConnectorImages
+from pipelines.airbyte_ci.connectors.consts import CONNECTOR_TEST_STEP_ID
+from pipelines.airbyte_ci.connectors.context import ConnectorContext, PipelineContext
+from pipelines.airbyte_ci.connectors.generate_erd.dbml_assembler import Source, DbmlAssembler
+from pipelines.airbyte_ci.connectors.generate_erd.relationships import RelationshipsMerger, Relationships
+from pipelines.airbyte_ci.connectors.reports import Report
+from pipelines.consts import LOCAL_BUILD_PLATFORM
+from pipelines.helpers.connectors.command import run_connector_steps
+from pipelines.helpers.execution.run_steps import STEP_TREE, StepToRun
+from pipelines.models.steps import Step, StepResult, StepStatus
+from pydbml.renderer.dbml.default import DefaultDBMLRenderer
+
+if TYPE_CHECKING:
+    from anyio import Semaphore
+
+# TODO: pass secret in dagger?
+API_KEY = "API_KEY"
+genai.configure(api_key=API_KEY)
+
+# TODO: pass secret in dagger
+DBDOCS_TOKEN = "TOKEN"
+
+
+def _get_erd_folder(code_directory: Path) -> Path:
+    path = code_directory / "erd"
+    if not path.exists():
+        path.mkdir()
+    return path
+
+def _get_discovered_catalog_file(code_directory: Path) -> Path:
+    return _get_erd_folder(code_directory) / "discovered_catalog.json"
+
+def _get_estimated_relationships_file(code_directory: Path) -> Path:
+    return _get_erd_folder(code_directory) / "estimated_relationships.json"
+
+def _get_confirmed_relationships_file(code_directory: Path) -> Path:
+    return _get_erd_folder(code_directory) / "confirmed_relationships.json"
+
+def _get_dbml_file(code_directory: Path) -> Path:
+    return _get_erd_folder(code_directory) / "source.dbml"
+
+
+class GenerateErdSchema(Step):
+    context: ConnectorContext
+
+    title = "Generate ERD schema using Gemini LLM"
+
+    def __init__(self, context: PipelineContext) -> None:
+        super().__init__(context)
+        self._model = genai.GenerativeModel("gemini-1.5-flash")
+
+    async def _run(self, connector_to_discover: Container) -> StepResult:
+        connector = self.context.connector
+        python_path = connector.code_directory
+        file_path = Path(os.path.abspath(os.path.join(python_path)))
+        source_config_path_in_container = "/data/config.json"
+        config_secret = open(file_path / "secrets" / "config.json").read()
+        discover_output = (
+            await connector_to_discover.with_new_file(source_config_path_in_container, contents=config_secret)
+            .with_exec(["discover", "--config", source_config_path_in_container])
+            .stdout()
+        )
```

Comment:
This will not work unless the developers running the command has the connector secrets locally.
I would suggest fetching secrets from GSM instead like we do for running CAT.
We use the test secrets declared in the `metadata.yaml` file. 
You can access secrets we use for CAT with `self.context.get_secrets_for_step_id(CONNECTOR_TEST_STEP_ID.ACCEPTANCE)`.

I would ideally suggest that you set, in `metadata.yaml` a new  `erdGeneration` `suite` in `connectorTestSuitesOptions` which would declare configs stored in GSM that you'd like to use for `discover`. 

This is not blocking ATM if you're fine having to fetch connector secrets before running the erd generation, but this is something you'd have to consider if you want to generate ERD in CI. 

There's also a better way to access local connector secrets: 
```
self.context.local_secret_store.get_all_secrets()
```

Reply:
This is interesting. Can we rely on `config.json` always being there and use that? If not indeed we would need to have more information and we chan update the metadata to have a `erd` block that we can add more information

Reply:
So in order not to have to modify the metadata file yet, we will continue to assume the `secrets/config.json` file exists locally. However, we will use the `local_secret_store` to access the config